### PR TITLE
Adding package supportdata

### PR DESCRIPTION
**Pull request by castelao**
This is a tiny package used to support the data files required by other packages such as OceansDB and CoTeDe.

Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/supportdata/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: "https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz"
+  sha256: 606346ee39642a8395814b3a55bc9f9b8273414cc02311b45e4a44307272a51d
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
```
Comment (scopatz) [inline recipes/supportdata/meta.yaml:14]: This can be `noarch: python`

File: recipes/supportdata/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  home: "https://github.com/castelao/supportdata"
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: LICENSE.rst
+  summary: "Download support data for Python pacakges, like sample data for tests and binary databases."
+  dev_url: https://github.com/castelao/supportdata
+
+extra:
+  recipe-maintainers:
+    - your-github-id-here
```
Comment (scopatz) [inline recipes/supportdata/meta.yaml:41]: Please add your github id

Comment (castelao) [reply recipes/supportdata/meta.yaml]: I'm sorry about that.

File: recipes/supportdata/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+  noarch: python
+
+requirements:
+  host:
+    - filelock >=2.0
+    - pip
+    - python
+    - six >=1.5.2
```
Comment (marcelotrevisani) [inline recipes/supportdata/meta.yaml:22]: I believe here you just need ``pip`` and ``python``
```suggestion
    - pip
    - python
```

File: recipes/supportdata/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  run:
+    - filelock >=2.0
+    - python
+    - six >=1.5.2
+
+test:
+  imports:
+    - supportdata
```
Comment (marcelotrevisani) [inline recipes/supportdata/meta.yaml:30]: Can we add ``pip check`` here as well, please?
```suggestion
    - supportdata
  requires:
    - pip
  commands:
    - pip check
```

File: recipes/supportdata/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+about:
+  home: "https://github.com/castelao/supportdata"
+  license: BSD-3-Clause
+  license_family: BSD
+  license_file: LICENSE.rst
```
Comment (marcelotrevisani) [inline recipes/supportdata/meta.yaml:36]: The license file is already being packaged on ``pypi``, so you do not need to add it manually 
```suggestion
  license_file: LICENSE
```

You can also remove the license file in this PR

File: recipes/supportdata/LICENSE.rst [hunk 6]
```diff
@@ -0,0 +1,12 @@
+Copyright (c) 2015-2020, Guilherme Castelao
```
Comment (marcelotrevisani) [inline recipes/supportdata/LICENSE.rst:1]: you can remove this file as explained in https://github.com/conda-forge/staged-recipes/pull/12320/files#r466860511

**Pull request merged**